sympy/solvers/tests/test_solveset.py

```diff
@@ -932,6 +935,19 @@ def test_issue_9522():
     assert solveset(expr2, x, S.Reals) == EmptySet()
 
 
+def test_solvify():
+    x = Symbol('x')
+
+    assert solvify(x**2 + 10, x, S.Reals) == []
+    assert solvify(x**3 + 1, x, S.Complexes) == [-1, 1/2 - sqrt(3)*I/2,
+                                                 1/2 + sqrt(3)*I/2]
+    assert solvify(log(x), x, S.Reals) == [1]
+    assert solvify(cos(x), x, S.Reals) == [pi/2, 3*pi/2]
+    assert solvify(sin(x) + 1, x, S.Reals) == [3*pi/2]
+    assert solvify(sin(Abs(x)), x, S.Reals) is None
```

Comment:
> This also reminds us that we need better ways to extract solutions from nested intersections and unions.

Yes. We need to.
I think @aktech suggested implementing a wrapper class to do this.